lib/dao.js

```diff
@@ -205,9 +205,11 @@ DataAccessObject.create = function (data, options, cb) {
         }
         data[i] = results[i].result;
       }
-      cb(errors, data);
+      if (cb.promise) {
+        cb(null, data);
+      } else cb(errors, data);
```

Comment:
This does not look right, you are basically swallowing errors in the promise case.

In the original PR, I was suggesting to drop support for the array version of `create` in the promise mode. Later, I have realised that such change can break the existing code calling the array version of create with no callback.

Here is a better solution: when in the array and promise mode, build a single array where each item is either the model instance created or the error object when create failed. If there was at least one error, then reject the promise. If all models were created successfully, then resolve the promise with the array.

Please make sure this case is covered by unit-tests (I did not check the test code yet).

Thoughts?